dali/pipeline/data/types_test.cc

```diff
@@ -97,19 +97,32 @@ typedef ::testing::Types<uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t,
 
 TYPED_TEST_SUITE(TypesTest, TestTypes);
 
+IMPL_HAS_MEMBER(value);
+
+template <typename T, decltype(type2id<T>::value) = {}>
+void CheckBuiltinType(const TypeInfo *t) {
+  EXPECT_EQ(t->id(), type2id<T>::value);
+  EXPECT_EQ(TypeTable::GetTypeId<T>(), type2id<T>::value);
+}
+
+template <typename T>
+void CheckBuiltinType(...) {}
+
+
```

Comment:
This passes in a single-DLL scenario, but it's a valid test anyway, so let's keep it.